conans/model/build_info.py

```diff
@@ -322,11 +332,43 @@ def _aggregated_paths(self, item):
             return paths
         paths = getattr(self._cpp_info, "%s_paths" % item)
         if self._cpp_info.components:
-            for component in self._cpp_info.components.values():
+            for component in self._get_sorted_components().values():
                 paths = self._merge_lists(paths, getattr(component, "%s_paths" % item))
         setattr(self, "_%s_paths" % item, paths)
         return paths
 
+    def _get_sorted_components(self):
+        """
+        Sort Components from most dependent one first to the less dependent one last
+        :return: List of sorted components
+        """
+        if not self._sorted_components:
+            if any([[require for require in comp.requires] for comp in
+                    self._cpp_info.components.values()]):
+                ordered = OrderedDict()
+                components = copy(self._cpp_info.components)
+                while len(ordered) != len(self._cpp_info.components):
+                    # Search next element to be processed
+                    for comp_name, comp in components.items():
+                        if comp_name in ordered:
+                            continue
+                        # check if all the deps are declared
+                        if not all([dep in components for dep in comp.requires]):
+                            raise ConanException("Component '%s' declares a missing dependency" % comp_name)
+                        # check if component is not required and can be added to ordered
+                        if comp_name not in [require for dep in components.values() for require in
```

Comment:
Probably way simpler and faster to sort in the opposite order: check that the current components requires are None or all in ``ordered``.

Reply:
I implemented that logic before and it was a mess calculating the link order later. I'd like to keep it as is for now and improve the speed in the future if needed